build.gradle

```diff
@@ -961,6 +963,53 @@ project(':streams') {
   }
 }
 
+project(':streams:streams-scala') {
+  println "Building project 'streams-scala' with Scala version ${versions.scala}"
+  apply plugin: 'scala'
+  archivesBaseName = "kafka-streams-streams-scala"
+
+  // use experimental flag to enable SAM type conversions in compiler
+  if (versions.baseScala == '2.11') {
+    tasks.withType(ScalaCompile) {
+      scalaCompileOptions.additionalParameters += ["-Xexperimental"]
```

Comment:
I agree it looks concerning, I'll need to check what other potential features this brings in, unfortunately there's no way to be more specific about just enabling SAM type conversion AFAIK.

We could remove this flag, but we would need to desugar all the places where conversions occur.

Reply:
> I'm also wondering if we really want to use the same class names as the Java DSL, this seems like it would be confusing to use in an IDE with code-completion. What do you think about using ScalaKStream, etc. instead? This would also get around the desire to rename imports.

Hi @vvcephei - regarding the above, we had a discussion on the 2 approaches on this thread only and the universal suggestion was to use the same name across Scala and Java APIs. In fact in the initial version that we posted, we had different names (`KStream` / `KStreamS`). The reasoning of using the same names is that the renaming of imports in the user code needs to be done only very occasionally when we mix usage of Scala and Java APIs.

Reply:
@vvcephei There's no list of what's brought in with the experimental flag, unless you grep the compiler code.  The purpose of the SAM type conversion feature in 2.11 was only to get early feedback and only properly finished in 2.12.  As its name suggests it's not meant to be used in production code.  Since Kafka is still targeting 2.11 it makes sense to not include this flag to build a releasable artifact.  I'll remove the flag and desugar the conversions.

Reply:
@debasishg Thanks for the explanation, I missed that discussion. Sorry to bring it back up!

@seglo Thanks! Am I correct in thinking that only affects our code, and not our users' code? I.e., if they are using 2.12, they can pass in lambdas as arguments, right?